### PR TITLE
ODS Loader

### PR DESCRIPTION
**Pull request by CharlesRendle**
Simple ODS loader - functionality is much of the same as in gss-utils when an ods file is loaded into databaker, it is first transferred into an xls file (xlrd dependency).

Uses pandas and odfpy as the engine for reading the ods contents into dataframes and then writing to an xlsx file.
Adds 6 seconds to the runtime of the whole test suite which seems abnormally long but this is also in line with the other `load_x_file_types.features`

Something to look at in the review: Where the new xlsx file ends up. `ods_as_xlsx.xlsx` when created, is put in the root databaker directory. I am not sure where would be best as we want it to be the same as the path from `input_file_obj`

### CODE REVIEW

File: databaker/framework.py [hunk 1]
```diff
@@ -35,6 +37,15 @@ def loadxlstabs(input, sheetids="*", verbose=True):
...
         elif str(input_file_name).endswith(".xls"):
             tableset = XLSTableSet(filename=input_file_name, fileobj=input_file_obj)
+        elif str(input_file_name).endswith(".ods"):
+            df_dict = pd.read_excel(str(input_file_name), engine="odf", sheet_name=None)
+            from pandas import ExcelWriter
+            w = ExcelWriter("ods_as_xlsx.xlsx")
+            for key in df_dict.keys():
+                ods_as_xlsx = df_dict[key].to_excel(w, sheet_name=key)
+                w.save()
+            tableset = XLSXTableSet(filename="ods_as_xlsx.xlsx", fileobj=ods_as_xlsx)
```
Comment (mikeAdamss) [inline databaker/framework.py]: we're only really gonna want to pass fileobj (you probably don't want the `.w.save()` either. that way we get the data but we're only converting the file in memory and avoid writing anything to disk (keeps it nice and seemless from a user perspective)

so just `tableset = XLSXTableSet(fileobj=ods_as_xlsx)` is fine.

File: features/loadods.feature [hunk 2]
```diff
@@ -0,0 +1,16 @@
+Feature: Load ods files
+  I want to load ods files into databaker
+
+  Scenario: Load ods file 1
+    Given we load a file named "2018internationaltradeinservicesdatatables.ods"
```
Comment (mikeAdamss) [inline features/loadods.feature:18]: use this: https://github.com/GSS-Cogs/databaker/blob/bc24db9a106a5a03108a99e5e952056104257b95/features/steps/load_xls.py#L53 , should let you test the read from file object (see other comment) is working.

File: features/loadods.feature [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+    Then we confirm the names of the loaded tabs are equal to:
+    """
+    ["Notes", "Index", "1. NUTS1, industry", "2. NUTS1, industry, destination",
+    "3. NUTS2, industry", "4. NUTS2, industry, destination", "5. NUTS3, destination",
+    "6. City Region, industry", "7. City Region, industry, dest.", "8. Travel", "9. Tidy format"]
+    """
+    And the output "some_name" should be equal to:
```
Comment (mikeAdamss) [inline features/loadods.feature]: this cant be right 😆 , needs a look but I'm guessing that variable is never used. Should probably clean up whatever this is (I'm betting a leftover from my initial hacking).

File: databaker/framework.py [hunk 4]
```diff
@@ -35,6 +37,15 @@ def loadxlstabs(input, sheetids="*", verbose=True):
             tableset = XLSXTableSet(filename=input_file_name, fileobj=input_file_obj)
         elif str(input_file_name).endswith(".xls"):
             tableset = XLSTableSet(filename=input_file_name, fileobj=input_file_obj)
+        elif str(input_file_name).endswith(".ods"):
+            df_dict = pd.read_excel(str(input_file_name), engine="odf", sheet_name=None)
```
Comment (mikeAdamss) [inline databaker/framework.py]: need to check if this works with a file object, so if I do:

```
with open("myfile.ods", "r") as f:
        loadxlstabs(f)
```

it needs to support this (this is principally how data will be coming in from gssutils so we need to know). I've added a comment to the feature for testing this.

**Review by mikeAdamss — COMMENTED** [review]
looking good (nice one). A few comments around file object stuff that need unpicking but I think we're almost there.

File: features/loadods.feature (hunk 3 above)
Comment (CharlesRendle) [reply features/loadods.feature]: Reworded this step and removed the variable.

File: features/loadods.feature (hunk 2 above)
Comment (CharlesRendle) [reply features/loadods.feature:18]: Added a duplicate test which keeps this `"Given"`.

File: databaker/framework.py (hunk 4 above)
Comment (CharlesRendle) [reply databaker/framework.py]: Added a test which uses this as a template for the step definition - leads to `context.tabs = ` and then follows the same structure as existing tests.

**Comment by CharlesRendle** [comment]
Seeing a test has just failed - will have to look at it in the morning!

**Comment by mikeAdamss** [comment]
broke this somehow 😄 , will fix at some point.

note to self/Charles - we need to be able to turn off the fallback handling when testing. At the moment it's hard to know whether something is _really_ passing of failing.

**Pull request merged**